Improving sensitive hit creation for FT0 in Detector.cxx

zovarga opened the pull request:
- The Detector::ProcessHits(FairVolume* v) function had too many string comparisons which made it slow. The comparison was modified to be based on integer IDs instead.
- The calculation of some quantities (etot, enDep, parentID) were moved inside the relevant if() statement block (i.e. when the particle is a photon), so they are only calculated when necessary.

zovarga committed:
Improving sensitive hit creation for FT0 in Detector.cxx

- The Detector::ProcessHits(FairVolume* v) function had too many string comparisons which made it slow. The comparison was modified to be based on integer IDs instead.
- The calculation of some quantities (etot, enDep, parentID) were moved inside the relevant if() statement block (i.e. when the particle is a photon), so they are only calculated when necessary.

sawenzel commented on Detectors/FIT/FT0/simulation/src/Detector.cxx:
```diff
@@ -870,7 +870,11 @@ void Detector::defineFrameTransformations()
+  Int_t volID = fMC->CurrentVolID(copy);
+  Int_t volREG = fMC->VolId("0REG");
+  Int_t volTOP = fMC->VolId("0TOP");
+  Int_t volMTO = fMC->VolId("0MTO");
```
these should be cached as data members inside the Detector class to avoid calling `VolId` again and again.

sawenzel reviewed (changes requested):
Looks good. Please still address the comment concerning caching volume ids as data members.

zovarga committed:
Update FT0/simulation/include/FT0Simulation/Detector.h

The volume IDs have been defined.

zovarga committed:
Update FIT/FT0/simulation/src/Detector.cxx

The volume IDs have been added to Detector::InitializeO2Detector().

alibuild committed:
Please consider the following formatting changes

zovarga committed:
Merge pull request #2 from alibuild/alibot-cleanup-7961

Please consider the following formatting changes to #7961

zovarga commented:
Thank you for the review, I implemented the requested changes.

sawenzel commented on Detectors/FIT/FT0/simulation/include/FT0Simulation/Detector.h:
```diff
@@ -248,6 +248,11 @@ class Detector : public o2::base::DetImpl<Detector>
+  // Define volume IDs
+  int regVolID = -1;
+  int topVolID = -1;
+  int mtoVolID = -1;
```
please note that class member variables need to follow the `mFooBar` notation.

zovarga committed:
Updating name of volume IDs

The name of volume IDs in Detector.h has been updated to conform to the standard naming convention.

zovarga committed:
Update name of volume IDs in Detector.cxx

Updated the name of volume IDs to conform to the standard naming conventions.

sawenzel commented:
@ktf @TimoWilken : There is a weird alibuild error here:
```
++ alibuild-generate-module --bin --lib
/System/Volumes/Data/build/ci_checks/alice_ci_0/o2/sw/SPECS/osx_arm64/TBB/v2021.5.0-local1/TBB.sh: line 13: alibuild-generate-module: command not found
```
I have seen this locally as well this morning.

TimoWilken commented:
Looks like TBB should depend on alibuild-recipe-tools!

sawenzel commented:
@TimoWilken : Right. But why did all the tests in alidist pass? It appears to be a bit platform-dependent at least.

TimoWilken commented:
Maybe the build order changed, and something else that depends on alibuild-recipe-tools was built before TBB. In that case, even though TBB doesn't explicitly depend on a-r-t, it would have it available. In this case, TBB was built before anything that pulls in a-r-t, so a-r-t wasn't available.

(Edited to add: this may well be platform-dependent, as using packages from the system would change the build graph and therefore the build order.)
